lib/src/stream_completer.dart

```diff
@@ -173,8 +173,9 @@ class _CompleterStream<T> extends Stream<T> {
     if (_controller == null) {
       _createController();
     }
-    _sourceStream = _controller.stream; // Mark stream as set.
-    _controller.close();
+    var controller = _controller!;
+    _sourceStream = controller.stream; // Mark stream as set.
+    controller.close();
```

Comment:
[optional] Could refactor so that `_createController` doesn't  assign the field, but returns it.

Then

```dart
var controller = _controller ??= _createController();
_sourceStream = controller.stream;
controller.close();
```

Reply:
done